### PR TITLE
test: implement Molecule based testing

### LINKED ISSUE #16: add Molecule-based testing
Currently, there's no easy way to deploy and test the roles in a local and confined environment. Molecule has been proven to be quite useful for just that.

The implementation will contain two Molecule scenarios, both of which will be based von VM technology (VirtualBox and libvirt).

### PR DESCRIPTION
**Pull request by DasFaultier**
This commit includes:
- Molecule scenarios for Ubuntu 22.04 and Alma Linux 9 with VirtualBox and libvirt backing for each of them
- yamllint config and checks
- ansible-lint and checks
- fixes for any warnings and errors that came up during Molecule test runs

This feature is also the base for building a convenient CI pipeline in GitHub for checking new commits.

### CODE REVIEW

**Comment by robpomeroy** [comment]
Jörg, this is amazing, thank you. I'll review this at my earliest opportunity.

Would you mind sending me your email address, so we can correspond outside GitHub? Pick a method for contacting me [here](https://pomeroy.me/about/) - the contact form at the bottom goes to my "normal" email account. Thanks!

@DasFaultier

File: main.yml [hunk 1]
```diff
@@ -66,7 +66,7 @@
     # dconf is used to lock down the desktop
     - role: dconf
       tags: [dconf, desktop]
-    - role: gui-packages
-      tags: [gui-packages, desktop]
+    - role: gui_packages
```
Comment (robpomeroy) [inline main.yml:69]: What's the reason for the switch to underscores? I haven't come across a clear convention for this.

Comment (DasFaultier) [reply main.yml:69]: The Ansible Doc (https://galaxy.ansible.com/docs/contributing/creating_role.html#role-names) says:

> Role names are limited to lowercase word characters (i.e., a-z, 0-9) and ‘_’. No special characters are allowed, including ‘.’, ‘-‘, and space. (...)

But frankly, I actually followed the recommendation from ansible-lint, which is run with `molecule lint` or during every `molecule test` run.

**Comment by robpomeroy** [comment]
Jörg, I'm getting closer to Molecule working! My first run failed, with the following message in the error log:

> Stderr: VBoxManage.exe: error: RawFile#0 failed to create the raw output file /dev/null (VERR_PATH_NOT_FOUND)
> VBoxManage.exe: error: Details: code E_FAIL (0x80004005), component ConsoleWrap, interface IConsole

This is with the Ubuntu 22.04 runner. Molecule/Vagrant created the VMs successfully. Disabling the serial port in the VM (per [this comment](https://github.com/joelhandwell/ubuntu_vagrant_boxes/issues/1#issuecomment-292326347)) allows the VM to boot in VirtualBox, so I think that's the issue. Could you please amend the Molecule/Vagrant configuration? Let's assume that for the time being we won't be testing any serial peripherals.

Thanks!

File: main.yml (hunk 1 above)
Comment (robpomeroy) [reply main.yml:69]: Thank you for educating me! 🙂

**Comment by robpomeroy** [comment]
Approving and merging this. I will open the serial port problem as a separate issue.

**Review by robpomeroy — APPROVED** [review]

**Pull request merged**